cpp/tests/test_binary_serializer.cpp

```diff
@@ -0,0 +1,209 @@
+/* 
+* Copyright (C) 2023 German Aerospace Center (DLR-SC)
+*
+* Authors: Daniel Abele
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#include "boost/none.hpp"
+#include "boost/none_t.hpp"
+#include "matchers.h"
+#include "memilio/epidemiology/damping.h"
+#include "memilio/epidemiology/damping_sampling.h"
+#include "memilio/io/binary_serializer.h"
+#include "memilio/io/io.h"
+#include "memilio/utils/parameter_distributions.h"
+#include "memilio/utils/time_series.h"
+#include "memilio/utils/uncertain_value.h"
+#include "ode_secir/model.h"
+#include "ode_secir/parameter_space.h"
+#include "ode_secir/parameters.h"
+#include "gtest/gtest.h"
+#include <memory>
+
+namespace {
+    template<class T>
```

Comment:
- Do we need the emtpy namespace?
- Please format the whole file, on 3-4 occasions our clang-format was not applied

Reply:
empty namespace is only needed if you don't want your tests to fail randomly with bugs that are impossible to find!
https://stackoverflow.com/questions/357404/why-are-unnamed-namespaces-used-and-what-are-their-benefits